### DISCUSSION TITLE
How to calculate Agile / Tracker Average price per kWh

### DISCUSSION THREAD

**Discussion by anothermort** [BottlecapDave/HomeAssistant-OctopusEnergy#493] [Q&A]
I realise this might not be possible with the HA addon and I am still a HA newbie when doing clever things.

But is there a way of calculating Agile and/or tracker average used prices between two dates in HA?
i.e. Average Price: 10.41p/kWh between July and October
![2023-10-28 13_03_59-FYI_ Your energy statement - adamwalkerinuk@gmail com - Gmail — Mozilla Firefox](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/13786570/07be432b-f79c-4beb-b2b9-04362fbd3f6a)

Would be interesting to see the running average before bills or without using something like Guy Lipmans calculator for Agile [https://www.guylipman.com/octopus/](url)

Any help would be appreciated

**Comment by willenic** [comment]
The Energy Dashboard does that. I think in the next update you can also select more customised date ranges.
